sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/LRUCache.java

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+public class LRUCache<K, V> extends LinkedHashMap<K, V> {
```

Comment:
is this thread safe?

Reply:
This is not thread safe! and can throw ConcurrentModificationException. I tried adding thread safety but any additional synchronization added is slowing down the cache and resulting in much less perf. So in the current implementation I am just catching this exception and doing nothing, as we are good if one of the attempts to write succeeds at some point.

Reply:
Maybe I am too pragmatic - but do we really need an LRUCache? At least if my assumption above is correct that the size could be in the order of hundreds or thousands we could have simpler implementation like replacing the map with a new one if it ever exceeds the max. size or similar and use a thread-safe map implementation instead?

Reply:
I don't think we should be using a non-thread safe construct in a multi-threading fashion. I think relying on `ConcurrentModificationException` is not the correct pattern.

Reply:
As we are bounding the cache size at this point, we need a mechanism to make sure we have policy on when we replace the query in the cache and LRU works well in this case by caching repeatedly used queries. We may not be much interested in one off queries as caching/not caching would not make much of a difference for a query run very rarely